Assets/Scripts/Layers/LayerTypes/ObjectScatterLayerGameObject.cs

```diff
@@ -15,69 +17,115 @@ public enum FillType
+        public PolygonSelectionLayer polygonLayer;
+        private List<IPropertySectionInstantiator> propertySections = new();
+        private List<PolygonVisualisation> visualisations = new();
 
-        public void Initialize(GameObject originalObject, PolygonSelectionLayer polygon, List<LayerData> children)
+        private Bounds polygonBounds = new();
+        private SampleTexture sampleTexture;
+
+        private WorldTransform worldTransform;
+
+        private bool completedInitialization;
+        public LayerPropertyData PropertyData => settings;
+
+        private void Awake()
         {
-            this.originalObject = originalObject;
-            this.mesh = CombineHierarchicalMeshes(originalObject.transform);
-            this.material = originalObject.GetComponentInChildren<MeshRenderer>().material; //todo: make this work with multiple materials for hierarchical meshes?
-            this.material.enableInstancing = true;
+            toggleScatterPropertySectionInstantiator = GetComponent<ToggleScatterPropertySectionInstantiator>();
+            propertySections = new List<IPropertySectionInstantiator>() { toggleScatterPropertySectionInstantiator, this };
+        }
+        
+        public void Initialize(LayerGameObject originalObject, PolygonSelectionLayer polygon)
+        {
+            
+            foreach (var property in originalObject.LayerData.LayerProperties)
+            {
+                LayerData.AddProperty(property); //copy properties to be able to revert
+            }
+
+            InitializeScatterMesh(originalObject.PrefabIdentifier);
 
-            originalObject.SetActive(false); //todo: does this affect the WorldTransformShifter?
             polygonLayer = polygon;
+            
+            var existingScatterProperties = (ScatterGenerationSettingsPropertyData) originalObject.LayerData.LayerProperties.FirstOrDefault(p => p is ScatterGenerationSettingsPropertyData);
+            if(existingScatterProperties == null)
+                InitializeNewScatterProperties(originalObject.PrefabIdentifier, polygon.ShapeType);
+            else
+                LoadScatterProperties(existingScatterProperties);
+            
+            propertySections = new List<IPropertySectionInstantiator>() { toggleScatterPropertySectionInstantiator, this };
 
-            toggleScatterPropertySectionInstantiator = GetComponent<ToggleScatterPropertySectionInstantiator>();
+            LayerData.SetParent(polygon);
 
-            if (!toggleScatterPropertySectionInstantiator)
-                toggleScatterPropertySectionInstantiator = gameObject.AddComponent<ToggleScatterPropertySectionInstantiator>();
+            RecalculatePolygonsAndSamplerTexture();
+            AddReScatterListeners();
 
-            settings = ScriptableObject.CreateInstance<ScatterGenerationSettings>();
+// #if UNITY_EDITOR
+//          gameObject.AddComponent<GridDebugger>();
```

Comment:
Is dit overgebleven debug comments?

Reply:
nee, deze stond er nog, ik kan die wel weghalen